src/main/java/org/jetbrains/kotlinx/lincheck/nvm/Persistent.kt

```diff
@@ -0,0 +1,71 @@
+/*-
+ * #%L
+ * Lincheck
+ * %%
+ * Copyright (C) 2019 - 2020 JetBrains s.r.o.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+package org.jetbrains.kotlinx.lincheck.nvm
+
+import org.jetbrains.kotlinx.lincheck.runner.RecoverableStateContainer
+
+
+/** Persistent reference emulates non-volatile memory variable with volatile cache. */
+class Persistent<T>() {
```

Comment:
Let's rename to `NonVolatileXxx`, where `Xxx` is `Reference`, `Int`, `Long`, etc.

Let's also add the methods like `compareAndSet` and `getAndSet` use the name like in `AtomicXxx` from kotlinx.atomicfu (using it under the hood)